Root webpage wont load

Hey Chris,

Awsome Project, excited to get it working and setup but I've had alot of trouble today trying to get it to work.

I run though the readme and followerd all the steps, only thing that didn't work for me was the upload via command prompt, I've been able to flash the esp32cam via platformio in vscode in any case so I dont think that matters too much

The westerner does start and I can get some some of the subpage ok, but when I try to go to the  root espcam.local/ or espcam.local/index.html I got a 404 error.

Any help on this would be appreciated

Thank you
Joe

Hi Chris,

Thank for getting back to me

Ive built it all again, and here is my output, with version numbers:

`
joe@Joeboy-Surbook:/mnt/c/Users/Joeboy/OneDrive/Home Automation/VSCode/VSCode - ESP32CAM LEDS$ python3 --version
Python 3.6.7
joe@Joeboy-Surbook:/mnt/c/Users/Joeboy/OneDrive/Home Automation/VSCode/VSCode - ESP32CAM LEDS$ python3 -m venv venv
joe@Joeboy-Surbook:/mnt/c/Users/Joeboy/OneDrive/Home Automation/VSCode/VSCode - ESP32CAM LEDS$ source ./venv/bin/activate
(venv) joe@Joeboy-Surbook:/mnt/c/Users/Joeboy/OneDrive/Home Automation/VSCode/VSCode - ESP32CAM LEDS$ pip install -r requirements.txtCollecting platformio (from -r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/af/7a/07e5fa67f08eb738bcea35c9a5aa862463e693de26e02d9f4d393a75f567/platformio-4.3.4.tar.gz
Collecting bottle<0.13 (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/e9/39/2bf3a1fd963e749cdbe5036a184eda8c37d8af25d1297d94b8b7aeec17c4/bottle-0.12.18-py3-none-any.whl
Collecting click<8,>=5 (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/d2/3d/fa76db83bf75c4f8d338c2fd15c8d33fdd7ad23a9b5e57eb6c5de26b430e/click-7.1.2-py2.py3-none-any.whl
Collecting colorama (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/c9/dc/45cdef1b4d119eb96316b3117e6d5708a08029992b2fee2c143c7a0a5cc5/colorama-0.4.3-py2.py3-none-any.whl
Collecting marshmallow>=2 (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/a6/11/f824483fcfc4164d733c0a242cc01f7e3b153561500a6a69bba217c9f0f6/marshmallow-3.7.1-py2.py3-none-any.whl
Collecting pyelftools<1,>=0.25 (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/bb/2f/bf41f3c3867d6707fa9b872658bb23088a64d0e522e8979f54c694b8cbe1/pyelftools-0.26-py2.py3-none-any.whl
Collecting pyserial!=3.3,<4,>=3 (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/0d/e4/2a744dd9e3be04a0c0907414e2a01a7c88bb3915cbe3c8cc06e209f59c30/pyserial-3.4-py2.py3-none-any.whl
Collecting requests<3,>=2.4.0 (from platformio->-r requirements.txt (line 1))
Collecting semantic_version<3,>=2.8.1 (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/a5/15/00ef3b7888a10363b7c402350eda3acf395ff05bebae312d1296e528516a/semantic_version-2.8.5-py2.py3-none-any.whl
Collecting tabulate<1,>=0.8.3 (from platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/c4/f4/770ae9385990f5a19a91431163d262182d3203662ea2b5739d0fcfc080f1/tabulate-0.8.7-py3-none-any.whl
Collecting chardet<4,>=3.0.2 (from requests<3,>=2.4.0->platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<3,>=2.5 (from requests<3,>=2.4.0->platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/a2/38/928ddce2273eaa564f6f50de919327bf3a00f091b5baba8dfa9460f3a8a8/idna-2.10-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests<3,>=2.4.0->platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/5e/c4/6c4fe722df5343c33226f0b4e0bb042e4dc13483228b4718baf286f86d87/certifi-2020.6.20-py2.py3-none-any.whl
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests<3,>=2.4.0->platformio->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/e1/e5/df302e8017440f111c11cc41a6b432838672f5a70aa29227bf58149dc72f/urllib3-1.25.9-py2.py3-none-any.whl
Building wheels for collected packages: platformio
  Running setup.py bdist_wheel for platformio ... error
  Complete output from command "/mnt/c/Users/Joeboy/OneDrive/Home Automation/VSCode/VSCode - ESP32CAM LEDS/venv/bin/python3" -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-jsxt_9uv/platformio/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp6zas8155pip-wheel- --python-tag 
cp36:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help

  error: invalid command 'bdist_wheel'

  ----------------------------------------
  Failed building wheel for platformio
  Running setup.py clean for platformio
Failed to build platformio
Installing collected packages: bottle, click, colorama, marshmallow, pyelftools, pyserial, chardet, idna, certifi, urllib3, requests, semantic-version, tabulate, platformio
  Running setup.py install for platformio ... done
Successfully installed bottle-0.12.18 certifi-2020.6.20 chardet-3.0.4 click-7.1.2 colorama-0.4.3 idna-2.10 marshmallow-3.7.1 platformio-4.3.4 pyelftools-0.26 pyserial-3.4 requests-2.24.0 semantic-version-2.8.5 tabulate-0.8.7 urllib3-1.25.9
(venv) joe@Joeboy-Surbook:/mnt/c/Users/Joeboy/OneDrive/Home Automation/VSCode/VSCode - ESP32CAM LEDS$ yarn --version
1.22.4
cd frontend && yarn
yarn install v1.22.4
[1/4] Resolving packages...
success Already up-to-date.
Done in 2.69s.
cd frontend && yarn build
yarn run v1.22.4
$ GENERATE_SOURCEMAP=false  react-app-rewired build
Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  56.8 KB  build/main.edb86cda.js
  453 B    build/main.4102b7b4.css

The project was built assuming it is hosted at /.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.
You may serve it with a static server:

  yarn global add serve
  serve -s build

Find out more about deployment here:

  bit.ly/CRA-deploy

Done in 132.78s.
cd frontend && yarn export
yarn run v1.22.4
$ rm -rf ../firmware/data && rm -f build/*manifest* && rm -f build/*LICENSE* && gzipper compress --level 9 --verbose ./build ../firmware/data
gzipper: Compression GZIP | level: 9
gzipper: Default output file format: [filename].[ext].[compressExt]
gzipper: File favicon.ico has been compressed 3.08 KB -> 2.91 KB (43.9662ms)
gzipper: File index.html has been compressed 609 B -> 356 B (31.336ms)
gzipper: File logo192.png has been compressed 5.22 KB -> 5.21 KB (38.2878ms)
gzipper: File logo512.png has been compressed 9.44 KB -> 9.43 KB (36.001ms)
gzipper: File main.4102b7b4.css has been compressed 850 B -> 453 B (29.5292ms)
gzipper: File main.edb86cda.js has been compressed 178.28 KB -> 56.8 KB (58.323ms)
gzipper: File robots.txt has been compressed 67 B -> 78 B (34.0236ms)
gzipper: File service-worker.js has been compressed 1.15 KB -> 683 B (35.2374ms)
gzipper: 8 files have been compressed. (1s 962.4901ms)
Done in 4.98s.
source ./venv/bin/activate && cd firmware && platformio run --target uploadfs
/bin/sh: 1: source: not found
Makefile:19: recipe for target 'upload_fs' failed
make: *** [upload_fs] Error 127
(venv) joe@Joeboy-Surbook:/mnt/c/Users/Joeboy/OneDrive/Home Automation/VSCode/VSCode - ESP32CAM LEDS$
 `

so based on that output it all seems to build just fine, other then the upload bit which i can do manually

It looks like the upload_fs failed because platformio didn't install.

```
Makefile:19: recipe for target 'upload_fs' failed
make: *** [upload_fs] Error 127
```

Since it did build the actual files you can do it manually from the vscode plugin:

![Screenshot 2020-07-21 at 11 03 19](https://user-images.githubusercontent.com/706944/88041289-180ce600-cb42-11ea-9b53-620fc340ef0e.png)
![Screenshot 2020-07-21 at 11 03 25](https://user-images.githubusercontent.com/706944/88041304-1cd19a00-cb42-11ea-99ee-21bb0c439afc.png)
![Screenshot 2020-07-21 at 11 03 30](https://user-images.githubusercontent.com/706944/88041318-20fdb780-cb42-11ea-83d1-9a4feeb86f56.png)

--- a/Makefile
+++ b/Makefile
@@ -8,15 +8,15 @@ compress_frontend:
 	cd frontend && yarn export
 
 build_firmware:
-	source ./venv/bin/activate && cd firmware && platformio run
+	. ./venv/bin/activate && cd firmware && platformio run
 
 build: build_frontend build_firmware
 
 upload_firmware:
-	source ./venv/bin/activate && cd firmware && platformio run --target upload
+	. ./venv/bin/activate && cd firmware && platformio run --target upload
 
 upload_fs: build_frontend compress_frontend
-	source ./venv/bin/activate && cd firmware && platformio run --target uploadfs
+	. ./venv/bin/activate && cd firmware && platformio run --target uploadfs
 
 
 upload: upload_fs upload_firmware

--- a/frontend/src/api/ledsApi.tsx
+++ b/frontend/src/api/ledsApi.tsx
@@ -1,22 +1,3 @@
-// get a raw frame from the camera - 160x120 pixels greyscale
-export async function getFrame(host: string): Promise<Uint8Array> {
-  const response = await fetch(`${host}/raw_image`);
-  const blob = await response.blob();
-  const arrayBuffer = await (blob as any).arrayBuffer();
-  const uint8Array = new Uint8Array(arrayBuffer);
-  return uint8Array;
-}
-
-// clear all the leds
-export async function clear(host: string) {
-  await fetch(`${host}/clear`);
-}
-
-export interface ILedPosition {
-  x: number;
-  y: number;
-}
-
 function blobToArrayBuffer(blob: Blob): Promise<ArrayBuffer> {
   return new Promise((resolve, reject) => {
     let fr = new FileReader();
@@ -29,6 +10,25 @@ function blobToArrayBuffer(blob: Blob): Promise<ArrayBuffer> {
     };
     fr.readAsArrayBuffer(blob);
   });
+}
+
+// get a raw frame from the camera - 160x120 pixels greyscale
+export async function getFrame(host: string): Promise<Uint8Array> {
+  const response = await fetch(`${host}/raw_image`);
+  const blob = await response.blob();
+  const arrayBuffer = await blobToArrayBuffer(blob);
+  const uint8Array = new Uint8Array(arrayBuffer);
+  return uint8Array;
+}
+
+// clear all the leds
+export async function clear(host: string) {
+  await fetch(`${host}/clear`);
+}
+
+export interface ILedPosition {
+  x: number;
+  y: number;
 }
 
 export async function getLedPositions(host: string): Promise<ILedPosition[]> {

--- a/frontend/src/components/Camera.tsx
+++ b/frontend/src/components/Camera.tsx
@@ -21,7 +21,7 @@ export default function Camera({
         });
         const canPlayListener = () => {
           onVideoReady(video);
-          alert(`canPlay 1 ${video.videoWidth} x ${video.videoHeight}`);
+          //alert(`canPlay 1 ${video.videoWidth} x ${video.videoHeight}`);
           setVideoWidth(video.videoWidth);
           setVideoHeight(video.videoHeight);
           video.removeEventListener("canplay", canPlayListener);

--- a/frontend/src/imageProcessing/imageProcessing.tsx
+++ b/frontend/src/imageProcessing/imageProcessing.tsx
@@ -100,8 +100,8 @@ export async function calibrateUsingCamera(
   espHost: string,
   video: HTMLVideoElement
 ) {
-  const width = video.videoWidth / 2;
-  const height = video.videoHeight / 2;
+  const width = video.videoWidth;
+  const height = video.videoHeight;
   // create a canvas element to capture video stills
   const canvas = document.createElement("canvas");
   canvas.width = width;

